modules/module8/module8-14-probabilities_and_logistic_regression.qmd

```diff
@@ -0,0 +1,426 @@
+
+# Find the target class probabilities of the test set using predict_proba. 
+# Save this in an object named proba_y
+proba_y = pmk_search.predict_proba(X_test)
+
+# This next part has been done for you
+lr_probs = pd.DataFrame({
+             "Pokemon": test_df['name'],
+             "true y":y_test, 
+             "pred y": predicted_y.tolist(),
+             "prob_legend": proba_y[:, 1].tolist()})
+             
+# Take the dataframe lr_probs and sort them in descending order of the models confidence
+# in predicting legendary pokemon
+# Save this in an object named legend_sorted       
+legend_sorted = lr_probs.sort_values(by='prob_legend', ascending=False)
+legend_sorted
+```
+
+:::
+::::
+
+<br>
+
+<div id='mcq5'></div>
+<script>
+    generateQuiz(
+        'mcq5',
+        'Question 1',
+        'Which option is the closest to your Best C value?',
+        {
+        '33.47': 'Your model’s value for the best C is incorrect. Are you scoring correctly?',
+        '48.26': 'Your model’s value for the best C is incorrect. Are you scoring correctly?',
+        '68.16': '',
+        '82.75': 'Your model’s value for the best C is incorrect. Are you scoring correctly?',
+        },
+        '68.16',
+    );
+</script>
+
+<div id='mcq6'></div>
+<script>
+    generateQuiz(
+        'mcq6',
+        'Question 2',
+        'Which option is the closest to your Best f1 score?',
+        {
+        '0.44': 'Your model’s f1 score is incorrect. Are you scoring correctly?',
+        '0.58': 'Your model’s f1 score is incorrect. Are you scoring correctly?',
+        '0.67': '',
+        '0.70': 'Your model’s f1 score is incorrect. Are you scoring correctly?',
+        },
+        '0.67',
+    );
+</script>
+
+<div id='mcq7'></div>
+<script>
+    generateQuiz(
+        'mcq7',
+        'Question 3',
+        'Which of the following Pokémon is the model confident in classifying it as "legendary"?',
+        {
+        'Cubchoo': 'Are you sorting <code>lr_probs</code> in descending order and looking at the top of the dataframe?',
+        'Lugia': '',
+        'Skitty': 'Are you sorting <code>lr_probs</code> in descending order and looking at the top of the dataframe?',
+        'Starmie': 'Are you sorting <code>lr_probs</code> in descending order and looking at the top of the dataframe?',
+        },
+        'Lugia',
+    );
+</script>
+
+<div id='mcq8'></div>
+<script>
+    generateQuiz(
+        'mcq8',
+        'Question 4',
+        'Which of the following Pokémon is the model confident in classifying it as  "not legendary"?',
+        {
+        'Uxie': 'Are you sorting <code>lr_probs</code> in descending order and looking at the bottom of the dataframe?',
+        'Zygarde': 'Are you sorting <code>lr_probs</code> in descending order and looking at the bottom of the dataframe?',
+        'Mandibuzz': 'Are you sorting <code>lr_probs</code> in descending order and looking at the bottom of the dataframe?',
+        'Diglett': '',
```

Comment:
I don't see diglett in the output data frame. Could you either change the name to another pokemon or print more rows to show diglett?

<img width="984" height="481" alt="image" src="https://github.com/user-attachments/assets/ef963ec1-1742-4a1d-ada6-1cce75606884" />